roiextractors/segmentationextractor.py

```diff
@@ -215,25 +154,66 @@ def get_image_size(self) -> ArrayType:
         """
         pass
 
-    def get_traces(self, roi_ids=None, start_frame=None, end_frame=None, name='Fluorescence'):
+    def get_traces(self, roi_ids=None, start_frame=None, end_frame=None, name='raw'):
         """
         Return RoiResponseSeries
         Returns
         -------
         traces: array_like
             2-D array (ROI x timepoints)
         """
-        if name not in self._roi_response_dict:
-            raise ValueError(f'traces for {name} not found, enter one of {list(self._roi_response_dict.keys())}')
+        if name not in self.get_traces_dict():
+            raise ValueError(f'traces for {name} not found, enter one of {list(self.get_traces_dict().keys())}')
         if roi_ids is None:
             roi_idx_ = range(self.get_num_rois())
         else:
             roi_idx = [np.where(np.array(i) == self.get_roi_ids())[0] for i in roi_ids]
             ele = [i for i, j in enumerate(roi_idx) if j.size == 0]
             roi_idx_ = [j[0] for i, j in enumerate(roi_idx) if i not in ele]
-        traces = self._roi_response_dict.get(name)
+        traces = self.get_traces_dict().get(name)
         return np.array([traces[int(i), start_frame:end_frame] for i in roi_idx_])
 
+    def get_traces_dict(self):
+        """
+        Returns traces as a dictionary with key as the name of the ROiResponseSeries
+        Returns
+        -------
+        _roi_response_dict: dict
+            dictionary with key, values representing different types of RoiResponseSeries
+            Flourescence, Neuropil, Deconvolved, Background etc
+        """
+        return deepcopy(dict(raw=np.array(self._roi_response_raw),
+                             dff=np.array(self._roi_response_dff),
+                             neuropil=np.array(self._roi_response_neuropil),
+                             deconvolved=np.array(self._roi_response_deconvolved)))
+
+    def get_images_dict(self):
+        """
+        Returns traces as a dictionary with key as the name of the ROiResponseSeries
+        Returns
+        -------
+        _roi_response_dict: dict
+            dictionary with key, values representing different types of Images used in segmentation:
+            Mean, Correlation image
+        """
+        return deepcopy(dict(mean=self._image_mean,
```

Comment:
I don't think you need a `deepcopy` here

Reply:
it does seem to make a difference: i tested it by manually changing the returned image_array (`get_images_dict()['mean']`) (without deepcopy) and it also changed the internal attribute (`self._image_mean`). With deepcopy this override does not alter the internal attribute.